Lombiq.Hosting.Tenants.IdleTenantManagement/Services/IdleShutdownTask.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Lombiq.Hosting.Tenants.IdleTenantManagement.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.BackgroundTasks;
+using OrchardCore.Entities;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Models;
+using OrchardCore.Modules;
+using OrchardCore.Settings;
+
+namespace Lombiq.Hosting.Tenants.IdleTenantManagement.Services;
+
+[BackgroundTask(Schedule = "* * * * *", Description = "Disable Idle Tenants.")]
+public class IdleShutdownTask : IBackgroundTask
```

Comment:
This is shutting down the current tenant, with the max idle minutes settings coming from the current tenant's settings. However, we don't want a tenant to be able to manage its own shutdown configuration (think DotNest).

The bg task can run on the tenant (but then on DotNest it needs to be always enabled) but the configuration should not be possible to edit there.